Podman quadlet

DiogoLobo122 opened the pull request.

DiogoLobo122 committed:
add branch for podman quadlet roles

DiogoLobo122 committed:
fix volume variable

DiogoLobo122 committed:
templating quadlet files

jhoekx commented on timeseer/podman/roles/timeseer/tasks/main.yml:
```diff
@@ -23,6 +23,16 @@
+      - "{{ timeseer_config_dir }}:/usr/src/app/config/:ro"
+
 - name: Create a podman network
   containers.podman.podman_network:
```
Do we need to create this in a file as well?

jhoekx commented on timeseer/podman/roles/timeseer/tasks/main.yml:
```diff
@@ -43,25 +53,41 @@
   when: timeseer_download_demo_data
 
-- name: Run Timeseer in a podman container
+- name: Create timeseer quadlet container
```
I think we should be able to skip the usage of the module when we create files.

jhoekx commented on timeseer/podman/roles/timeseer/tasks/main.yml:
```diff
@@ -43,25 +53,41 @@
+- name: Template configuration files
+  ansible.builtin.template:
+    src: "{{ item }}"
+    dest: "/etc/containers/systemd/ts-container.container"
```
To be consistent, the file should be called `timeseer.container`.

jhoekx commented on timeseer/podman/roles/timeseer/tasks/main.yml:
```diff
@@ -43,25 +53,41 @@
+    - Restart timeseer service
+
+- name: Flush Handlers to apply changes immediately
+  ansible.builtin.meta: flush_handlers
```
We should avoid `flush_handlers` in roles.

Use `timeseer_j2 is changed` to reload the systemd daemon conditionally.

jhoekx commented on timeseer/podman/roles/timeseer/handlers/main.yml:
```diff
@@ -0,0 +1,17 @@
+  when: daemon_reload | default(false) | bool
+
+- name: Restart timeseer service
+  ansible.builtin.systemd:
```
This should be conditional on the start of the service, to avoid an unneeded restart right after starting.

jhoekx commented on timeseer/podman/roles/timeseer/templates/timeseer-container.quadlet.j2:
```diff
@@ -0,0 +1,27 @@
+ContainerName=timeseer
+Image={{ timeseer_image }}
+User=1001
+Network=timeseer
```
```suggestion
Network={{ timeseer_network }}
```

jhoekx commented on timeseer/podman/roles/timeseer/templates/timeseer-container.quadlet.j2:
```diff
@@ -0,0 +1,27 @@
+{% for volume in timeseer_volume_list %}
+Volume={{ volume }}
+{% endfor %}
+PublishPort={{ timeseer_ports }}
```
If the variable is plural, I would expect a for loop over the variable that adds multiple `PublishPort` statement.

jhoekx commented on timeseer/podman/roles/timeseer/templates/timeseer-container.quadlet.j2:
```diff
@@ -0,0 +1,27 @@
+[Install]
+WantedBy=network-online.target
+
+
```
```suggestion
```

jhoekx commented on timeseer/podman/roles/traefik/templates/traefik-container.quadlet.j2:
```diff
@@ -0,0 +1,27 @@
+
+
+
+
```
```suggestion
```

jhoekx commented on timeseer/podman/roles/timeseer_reverse_proxy/templates/timeseer-reverse-proxy-container.quadlet.j2:
```diff
@@ -0,0 +1,26 @@
+[Install]
+WantedBy=network-online.target
+
+
```
```suggestion
```

jhoekx commented on timeseer/podman/roles/timeseer_data_service/templates/timeseer-data-service-container.quadlet.j2:
```diff
@@ -0,0 +1,31 @@
+
+
+
+
```
```suggestion
```

jhoekx commented on timeseer/podman/roles/timeseer_data_service/templates/timeseer-data-service-container.quadlet.j2:
```diff
@@ -0,0 +1,31 @@
+Wants=network.target
+After=network.target
+
+
```
```suggestion
```

jhoekx commented on timeseer/podman/roles/timeseer_data_service/templates/timeseer-data-service-container.quadlet.j2:
```diff
@@ -0,0 +1,31 @@
+ContainerName=timeseer-data-service
+Environment=RUST_LOG=info
+Exec=/usr/bin/timeseer-data-service
+LogDriver=journald
```
The order of the statements here is inconsistent with the previous one. This is confusing to people editing multiple quadlet files.

DiogoLobo122 committed:
Refactor reload management and add network file

- Remove unnecessary code
- Add network file

DiogoLobo122 committed:
change ports to list

jhoekx commented on timeseer/podman/roles/timeseer/tasks/main.yml:
```diff
@@ -43,25 +49,28 @@
+    daemon_reload: true
+  when: timeseer_container.changed # noqa: no-handler
+
+- name: Check if timeseer service is enabled
```
```suggestion
- name: Start and Enable Timeseer
```

jhoekx commented on timeseer/podman/roles/timeseer/handlers/main.yml:
```diff
@@ -0,0 +1,17 @@
+
+- name: Restart timeseer service
+  ansible.builtin.systemd:
+    name: ts-container
```
This name does not correspond to the name of the container.


```suggestion
    name: timeseer
```

jhoekx commented on timeseer/podman/roles/timeseer/handlers/main.yml:
```diff
@@ -0,0 +1,17 @@
+- name: Restart timeseer service
+  ansible.builtin.systemd:
+    name: ts-container
+    state: restarted
```
This should be conditional on `timeseer_service_status`.

jhoekx commented on timeseer/podman/roles/timeseer/handlers/main.yml:
```diff
@@ -0,0 +1,17 @@
+- name: Signal daemon Restart
+  ansible.builtin.set_fact:
+    daemon_reload: true
+  when: item.changed
+  loop: "{{ timeseer_j2.results }}"
+  loop_control:
+    label: "{{ item.dest }}"
+
+- name: Reload systemd daemon
+  ansible.builtin.systemd:
+    daemon_reload: "{{ daemon_reload }}"
+  when: daemon_reload | default(false) | bool
+
```
These handlers are unused.

```suggestion
```

jhoekx commented on timeseer/podman/roles/timeseer_data_service/tasks/main.yml:
```diff
@@ -15,27 +15,40 @@
+    src: "{{ item }}"
+    dest: "/etc/containers/systemd/timeseer-data-service.container"
+    mode: "0644"
+  loop:
```
This does not need to be a loop.

jhoekx commented on timeseer/podman/roles/timeseer_data_service/tasks/main.yml:
```diff
@@ -15,27 +15,40 @@
+  register: timeseer_data_service_container
+  notify:
+    - Restart timeseer-data-service service
+  when: timeseer_data_service_enable
```
It's cleaner and less error-prone to either have a block in `main.yml` with this condition or have a conditional include.

jhoekx replied:
The idea was not that there is one task in an include, but to have all tasks in this file moved to that separate file. They all should only run when the data service is enabled. That's why the one block here is also a valid option.

jhoekx commented on timeseer/podman/roles/timeseer_data_service/templates/timeseer-data-service-container.quadlet:
```diff
@@ -0,0 +1,25 @@
+PodmanArgs=--log-opt tag={{ timeseer_data_service_journal_tag }}
+Pull={{ timeseer_data_service_image_pull }}
+PodmanArgs=--tty
+PublishPort={{ timeseer_data_service_ports }}
```
Is this not a list?

Do we even need to publish the port?

jhoekx commented on timeseer/podman/roles/traefik/tasks/main.yml:
```diff
@@ -23,41 +23,50 @@
+- name: Reload systemd daemon to ensure traefik container is known
+  ansible.builtin.systemd:
+    daemon_reload: true
+  when: traefik_config.changed # noqa: no-handler
```
There's no need to reload systemd when the Traefik configuration changes. It's best to split up the templating task in two tasks: one for the configuration, another for the container.

jhoekx reviewed (changes requested):
I did not make the same comment twice, but some apply to multiple roles.

DiogoLobo122 committed:
add documentantion for quadlet

DiogoLobo122 committed:
remove double statement

DiogoLobo122 committed:
Update podman deployment docs
- Clarified usage of Quadlet files and systemctl commands.
- Provided examples for starting, stopping, and restarting the container.

DiogoLobo122 committed:
remove unused handlers, correct missing ones

DiogoLobo122 committed:
add conditional for port in the timeseer-data-service quadlet file
  - change variable port to a list

jhoekx commented on timeseer/podman/roles/timeseer_data_service/tasks/main.yml:
```diff
@@ -15,27 +15,32 @@
       - "{{ timeseer_data_service_config_dir }}:/usr/src/app/Timeseer-data-service.toml"
-    volume_list: "{{ base_volumes + (custom_toml_volume if timeseer_data_service_config_dir | default('') | trim != '' else []) }}"
+
+- name: Prepend 'systemd-' to network
```
Why is this not required for the network in the `timeseer` role?

DiogoLobo122 replied:
likely because the network configuration is already being handled within the quadlet file itself

`[Network]
Label={{ timeseer_network }}`


This specifies the network for the container directly in the configuration file. By doing so, the network is created and labeled correctly without needing additional manipulation in the playbook.


If you add the [Network] section to multiple quadlet files, it can lead to the creation of multiple networks i guess

jhoekx replied:
"likely", "guess" seems to indicate that the issue is not fully understood.

Can you point to the podman documentation where they explain that for the second container in a network, the network should be prefixed?

jhoekx replied:
Shouldn't `Network=timeseer.network` be used in the quadlet file as is done in the Timeseer container? Alternatively, the name of the `.network` file could be templated as well.

DiogoLobo122 committed:
correct task name

DiogoLobo122 committed:
Refactor network logic and improve configurability

- Remove playbook logic that pre-appends the network name.
- Add `.network` suffix to user-defined network variables.
- Allow users to define the network name.
- Move code from `main.yml` to `timeseer_data_service.yml` for better organization.

This closes #14.

DiogoLobo122 committed:
change to list
